experimental/adaptive-cards-2/Catering/Bots/CateringBot.cs

```diff
@@ -0,0 +1,265 @@
+                    cardOptions.option = cardOptions.custom;
+                }
+
+                user.Lunch.Drink = cardOptions.option;
+            }
+
+            AdaptiveCardInvokeResponse responseBody = null;
+            switch ((Card)cardOptions.nextCardToSend)
+            {
+                case Card.Drink:
+                    responseBody = DrinkCardResponse();
+                    break;
+                case Card.Entre:
+                    responseBody = EntreCardResponse();
+                    break;
+                case Card.Review:
+                    responseBody = ReviewCardResponse(user);
+                    break;
+                case Card.ReviewAll:
+                    var latestOrders = await _cateringDb.GetRecentOrdersAsync();
+                    responseBody = RecentOrdersCardResponse(latestOrders.Items);
+                    break;
+                case Card.Confirmation:
+                    await _cateringDb.UpsertOrderAsync(user);
+                    responseBody = ConfirmationCardResponse();
+                    break;
+                default:
+                    throw new NotImplementedException("No card matches that nextCardToSend.");
+            }
+
+            return responseBody;
+        }
+
+        private static InvokeResponse CreateInvokeResponse(HttpStatusCode statusCode, object body = null)
+        {
+            return new InvokeResponse()
+            {
+                Status = (int)statusCode,
+                Body = body
+            };
+        }
+
+        private static async Task SendWelcomeMessageAsync(ITurnContext turnContext, CancellationToken cancellationToken)
+        {
+            foreach (var member in turnContext.Activity.MembersAdded)
+            {
+                if (member.Id != turnContext.Activity.Recipient.Id)
+                {
+                    var message = MessageFactory.Text(WelcomeText);
+                    await turnContext.SendActivityAsync(message, cancellationToken: cancellationToken);
+                    await turnContext.SendActivityAsync($"Type anything to see a card here, or type recents to see recent orders.");
+                }
+            }
+        }
+
+        #region Cards As InvokeResponses
+
+        private AdaptiveCardInvokeResponse CardResponse(string cardFileName)
+        {
+            return new AdaptiveCardInvokeResponse()
+            {
+                StatusCode = 200,
+                Type = AdaptiveCard.ContentType,
+                Value = new CardResource(cardFileName).AsJObject()
+            };
+        }
+
+        private AdaptiveCardInvokeResponse CardResponse<T>(string cardFileName, T data)
+        {
+            return new AdaptiveCardInvokeResponse()
+            {
+                StatusCode = 200,
+                Type = AdaptiveCard.ContentType,
+                Value = new CardResource(cardFileName).AsJObject(data)
+            };
+        }
+
+        private AdaptiveCardInvokeResponse DrinkCardResponse()
```

Comment:
I would recommend removing these one line private helpers as they don’t add much, EntreCardResponse is only called once in this file.

Reply:
I moved them to be inline.